Update documents to reflect change to Microsoft Entra ID (#1665)
alan-turing-institute/data-safe-haven: docs/source/deployment/deploy_sre.md

```

````{error}

> If you see an error like the following when attempting to log in, it is likely that the Microsoft Entra application is not registered as an `ID token` provider.

Reviewer: ```suggestion
If you see an error like the following when attempting to log in, it is likely that the Microsoft Entra ID application is not registered as an `ID token` provider.
```

```{image} deploy_sre/guacamole_aad_idtoken_failure.png
:alt: AAD ID token failure
:align: center
```

> <details><summary><b>Register Microsoft Entra application</b></summary>

Reviewer: ```suggestion
<details><summary><b>Register Microsoft Entra ID application</b></summary>
```

![Microsoft Entra ID: one minute](https://img.shields.io/static/v1?style=for-the-badge&logo=microsoft-academic&label=Microsoft%20Entra%2
0ID&color=blue&message=one%20minute)

> - From the Azure portal, navigate to the ME-ID you have created.

Reviewer: ```suggestion
- From the Azure portal, navigate to the Microsoft Entra ID you have created.
```

- Navigate to `Microsoft Entra ID > App registrations`, and select the application called `Guacamole SRE <SRE ID>`.
- Click on `Authentication` on the left-hand sidebar
- Ensure that the `ID tokens` checkbox is ticked and click on the `Save` icon if you had to make any changes
  ```{image} deploy_sre/guacamole_aad_app_registration_idtoken.png

>   :alt: ME app registration

Reviewer: ```suggestion
  :alt: Microsoft Entra app registration
```
